tests/backend-storage/src/test/ServerStorage.test.ts

```diff
@@ -304,6 +304,49 @@ describe(`${ServerStorage.name}: ${serverStorage.constructor.name}`, () => {
     });
   });
 
+  describe(`${serverStorage.listDirectories.name}()`, () => {
+    it("should list directories", async () => {
+      const testDirectory1: TestRemoteDirectory =
+        await testDirectoryManager.createNew();
+      const testDirectory2: TestRemoteDirectory =
+        await testDirectoryManager.createNew();
+      const testDirectory3: TestRemoteDirectory =
+        await testDirectoryManager.createNew();
+      await testDirectory1.uploadFile(
+        { objectName: "reference1" },
+        undefined,
+        undefined
+      );
+      await testDirectory2.uploadFile(
+        { objectName: "reference2" },
+        undefined,
+        undefined
+      );
+      await testDirectory3.uploadFile(
+        { objectName: "reference3" },
+        undefined,
+        undefined
+      );
+      const queriedReferences = await serverStorage.listDirectories();
```

Comment:
What would be the behavior in the following cases:
- one of the directories (`testDirectory1` for example) does not have a file uploaded. is it listed?
- there's a hierarchy of directories:
```
root
-- dir1
  -- subDir1
    -- file.txt
  -- subDir2
    -- file.txt
```
is only `dir1` returned? Looking at the code I see that there is a filter for unique entries but it would be good to have a test case for that.

Reply:
> one of the directories (testDirectory1 for example) does not have a file uploaded. is it listed?

It depends on if we include empty directories or not.

> there's a hierarchy of directories:

From what I see here
[https://docs.aws.amazon.com/AmazonS3/latest/userguide/using-prefixes.html](url) it seems that I need to add a delimiter and I need to get baseDirectories from the CommonPrefix. Will change that. Thanks for noticing that!

Reply:
Does Azure include empty containers when querying?

Reply:
> Does Azure include empty containers when querying?

Yes it does. Ok so I'll include them.

Reply:
It would be good to have that case covered in a test

Reply:
I added it in a seperate test were I create a file in a subfolder : the case is like this

     root
       -- dir1
       -- dir2
         -- subDir1
            -- file.txt